python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -478,3 +479,40 @@ def cleanup(self):
     @staticmethod
     def bootstrap_config(cluster_config):
         return bootstrap_aws(cluster_config)
+
+    @staticmethod
+    def fillout_available_node_types_resources(
+            cluster_config: Dict[str, Any]) -> Dict[str, Any]:
+        """Fills out missing "resources" field for available_node_types."""
+        if "available_node_types" not in cluster_config:
+            return cluster_config
+        instances_list = boto3.client("ec2").describe_instance_types()[
+            "InstanceTypes"]
+        instances_dict = {
+            instance["InstanceType"]: instance
+            for instance in instances_list
+        }
+        available_node_types = cluster_config["available_node_types"]
+        for node_type in available_node_types:
+            instance_type = available_node_types[node_type]["node_config"][
+                "InstanceType"]
+            if "resources" not in available_node_types[node_type] and \
+                    instance_type in instances_dict:
+                cpus = instances_dict[instance_type]["VCpuInfo"][
+                    "DefaultVCpus"]
+                available_node_types[node_type]["resources"] = {"CPU": cpus}
+                gpus = instances_dict[instance_type].get("GpuInfo",
```

Comment:
Does this actually give the right GPU string values as in accelerators.py?

Reply:
lol it super conveniently appears to

Reply:
Yes.

Reply:
The available values are the ones available in the zone you are using.